DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/AllDelegates/AllDelegatesViewModel.cs

```diff
@@ -0,0 +1,95 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.AllDelegates
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Models.User;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
+
+    public class AllDelegatesViewModel : BaseSearchablePageViewModel
+    {
+        public AllDelegatesViewModel(
+            int centreId,
+            IEnumerable<DelegateUserCard> delegateUserCards,
+            IEnumerable<(int id, string name)> jobGroups,
+            CentreCustomPromptHelper centreCustomPromptHelper,
+            int page,
+            string? searchString,
+            string sortBy,
+            string sortDirection,
+            string? filterBy
+        ) : base(searchString, page, true, sortBy, sortDirection, filterBy)
+        {
+            CentreId = centreId;
+
+            var sortedItems = GenericSortingHelper.SortAllItems(
+                delegateUserCards.AsQueryable(),
+                sortBy,
+                sortDirection
+            );
+            var searchedItems = GenericSearchHelper.SearchItems(sortedItems, SearchString).ToList();

```

Comment:
I think it is best to remove the .ToList() from this line to prevent unnecessary enumeration.